File: ovvium-backend/src/main/java/com/ovvium/services/web/controller/bff/v1/transfer/response/rating/TotalRatingResponse.java
package com.ovvium.services.web.controller.bff.v1.transfer.response.rating;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.UUID;

@Getter
@RequiredArgsConstructor
public final class TotalRatingResponse {

	private final UUID productId;
	private final int rating;
	private final long total;
	private final float percentage;

}

File: ovvium-app/src/model/response/TotalRatingResponse.ts

export class TotalRatingResponse {
    productId: string;
    rating: number;
    total: number;
    percentage: number;

    constructor(totalRatingResponse: TotalRatingResponse = {} as TotalRatingResponse) {
        this.productId = totalRatingResponse.productId;
        this.rating = totalRatingResponse.rating;
        this.total = totalRatingResponse.total;
        this.percentage = totalRatingResponse.percentage;
    }
}